src/runtime_src/core/pcie/driver/linux/xocl/subdev/xvc.c

```diff
@@ -170,7 +171,21 @@ static long xvc_ioctl_helper(struct xocl_xvc *xvc, const void __user *arg)
 	}
 
 	total_bits = xvc_obj.length;
-	total_bytes = (total_bits + 7) >> 3;
+
+	/* Fixing integer overflow scenario */
+	if (total_bits + 7 >= UINT_MAX)
```

Comment:
i think, this wont work, at the end you are actually adding 7 to total_bits which can go out of bounds.
You should not be adding 7 to  check whether it is greater to UINT_MAX. You should be checking the other way,
You just need to add following check and keep other logic same

if (total_bits >= UINT_MAX - 7 )
        {
                total_bits = UINT_MAX - 7 ;
        }
        total_bytes = total_bits + 7 >> 3

Reply:
Hi @chvamshi-xilinx thanks for the review. I will fix the logic. Thank you